### PR TITLE
feat(engine): domain related updates & refactoring

### PR DESCRIPTION
**Pull request by weakish**
1. Rename subdomain (二级域名) to dev domain (开发域名),
    to avoid possbile confusions,
  and to clarify that they are intended to be used in dev phrase only.
2. Refactor quickstart guide, to reduce duplicated content
  between quickstart and webhosting guide.
  Also, impatient users can quickly start a project,
  less likely to get disappointed
  because they overlooked some basic points
  when skimming the lengthy webhosting guide.

### CODE REVIEW

File: views/dashboard_guide.md [hunk 1]
```diff
@@ -259,16 +259,16 @@ LeanCloud 不会把大家强制绑定到自己平台上，所以我们也提供
 云引擎实例的日志，则可以在 **应用日志** 和 **访问日志** 这里看到。
 云引擎实例运行状态的统计信息，则可以在 **统计** 看到，这里包含了云引擎运行过程中的调用次数、页面展现次数等核心指标的统计结果，你需要密切关注这里的指标，已决定是否需要对云引擎进行扩容。
 
-#### 云引擎中自定义二级域名
-云引擎默认为每个应用提供了一个二级域名 `<应用的域名>.{{engineDomain}}`，允许开发者将 web 应用部署到该域名之下运行，支持静态资源和动态请求服务。这个二级域名是可以修改的，你可以到 **设置 > Web 主机域名** 里面设置自定义的子域名。
+#### 云引擎中自定义开发域名
```
Comment (jysperm) [inline views/dashboard_guide.md]: 这里改成「云引擎开发域名」？现在已经不能自定义了（美国节点倒是还可以）

File: views/leanengine_cli.md [hunk 2]
```diff
@@ -299,7 +299,7 @@ $ lean deploy
 $ lean deploy -m 'Be more awesome! 这是定制的部署备注'
 ```
 
-部署之后可以通过 curl 命令来测试你的云引擎代码，或者访问你已设置的二级域名的测试地址 `stg-${应用的域名}.{{engineDomain}}`。
+部署之后可以通过 curl 命令来测试你的云引擎代码，或者访问你已设置的开发域名的测试地址 `stg-${应用的域名}.{{engineDomain}}`。
```
Comment (jysperm) [inline views/leanengine_cli.md]: `stg-${应用的域名}` 这个东西的标准说法是「预备环境」

File: views/unity_guide.md [hunk 3]
```diff
@@ -526,7 +526,7 @@ await file.DeleteAsync();
 
 ### 设置自定义文件域名
 
-LeanCloud 提供公用的二级域名来让开发者及其用户能够便捷地访问到存储在云端的文件。但由于受网络法规的管控与限制，我们无法 100% 保证该公用域名随时可用。因此，强烈建议开发者**使用自定义域名来访问自己的文件**，以避免公用域名不可用之时应用的体验会受到影响。请前往 [存储 > 设置 > 文件](/dashboard/storage.html?appid={{appid}}#/storage/conf) 设置自定义域名。
+LeanCloud 提供公用的开发域名来让开发者及其用户能够便捷地访问到存储在云端的文件。但由于受网络法规的管控与限制，我们无法 100% 保证该公用域名随时可用。因此，强烈建议开发者**使用自定义域名来访问自己的文件**，以避免公用域名不可用之时应用的体验会受到影响。请前往 [存储 > 设置 > 文件](/dashboard/storage.html?appid={{appid}}#/storage/conf) 设置自定义域名。
```
Comment (jysperm) [inline views/unity_guide.md]: 这里不对吧，这里说的是文件域名

File: views/dashboard_guide.md (hunk 1 above)
Comment (weakish) [reply views/dashboard_guide.md]: 改成了「云引擎绑定域名」

File: views/unity_guide.md (hunk 3 above)
Comment (weakish) [reply views/unity_guide.md]: 批量替换后检查时漏掉了这个，已修改。

File: views/leanengine_cli.md (hunk 2 above)
Comment (weakish) [reply views/leanengine_cli.md]: 已改

**Comment by weakish** [comment]
ping @jysperm 有空再看一下

**Pull request merged**